### PR TITLE
Update Readme file with more details for better on-boarding

### PR DESCRIPTION
**Pull request by kehiy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,29 @@
 [![Clippy check](https://github.com/gemwalletcom/core/actions/workflows/lint.yml/badge.svg)](https://github.com/gemwalletcom/core/actions/workflows/lint.yml)
 [![Gemstone CI](https://github.com/gemwalletcom/core/actions/workflows/ci-gemstone.yml/badge.svg)](https://github.com/gemwalletcom/core/actions/workflows/ci-gemstone.yml)
 
+# Gem Core Library
+
+The Gem Core Library is a Rust monorepo utilized within the Gem wallet. The Android and iOS versions of the wallet utilize this core as backend infrastructure, performing tasks such as push notifications, pricing, assets information, and more.
```
Comment (0xh3rman) [inline README.md:9]:
```suggestion
The Gem Core library is a Rust monorepo that serves as the backend infrastructure and cross platform libraries for the Gem Wallet on Android and iOS. It handles various tasks, including:

- Push notifications
- Pricing information
- Asset updating
- Shared modes using TypeShare
- Crypto primitives
- More
```

File: README.md [hunk 2]
```diff
@@ -4,6 +4,29 @@
...
+
+## Gemstone 
+
+The Gemstone is a cross-platform library for Swift and Kotlin.
+
+## Cryptography
+
+The cryptography for the Gem wallet is implemented using [trust-wallet-core](https://github.com/trustwallet/wallet-core).
```
Comment (0xh3rman) [inline README.md:17]:
```suggestion
The underlying cryptography / keystore for Gem Wallet is implemented using [Trust Wallet Core](https://github.com/trustwallet/wallet-core) for now.
```

File: README.md [hunk 3]
```diff
@@ -4,6 +4,29 @@
...
+
+```mermaid
+graph LR
+    client[gem client like: swift or kotlin]
```
Comment (0xh3rman) [inline README.md:21]:
```suggestion
    client[Gem Wallet (iOS/Android)]
```

File: README.md [hunk 4]
```diff
@@ -4,6 +4,29 @@
...
+    core-lib[gem wallet core library monorepo]
```
Comment (0xh3rman) [inline README.md:22]:
```suggestion
    core-lib[Gem Wallet Core]
```

File: README.md [hunk 5]
```diff
@@ -4,6 +4,29 @@
...
+    trust-core[trust wallet core]
```
Comment (0xh3rman) [inline README.md:23]:
```suggestion
    trust-core[Trust Wallet Core]
```

**Review by 0xh3rman — COMMENTED** [review]
Thanks for adding more to README, edit a little

**Comment by kehiy** [comment]
@0xh3rman your suggestions are committed.

**Review by 0xh3rman — APPROVED** [review]

**Pull request merged**